### ISSUE TITLE
Globals are always included for commonjs modules

### ISSUE THREAD

**Issue by TrySound** [cyco130/esbuild-plugin-polyfill-node#8]
Facing this in remix. `inject` in commonjs always init polyfills even when there is no usages of such globals. Maybe add support for at least disabling globals?
https://github.com/remix-run/remix/issues/6380

**Comment by MichaelDeBoey** [comment]
@cyco130 Is there anything I can do to help this getting included?

**Comment by cyco130** [comment]
It's on my todo, I"ll come to it in a day or two. I'll add an option to disable globals selectively and completely.

**Comment by cyco130** [comment]
Just released 0.3.0 which allows more fine-grained control over injected globals as well as `globals: false` to disable them completely.

**Comment by MichaelDeBoey** [comment]
@cyco130 It would be awesome if globals could be detected and auto-polyfilled if not present

**Comment by cyco130** [comment]
> @cyco130 It would be awesome if globals could be detected and auto-polyfilled if not present

You mean a runtime test? That would be out of scope for this project. But you could do something similar in your app's startup code maybe?

**Comment by MichaelDeBoey** [comment]
@cyco130 Yes, just like `esbuild-plugins-node-modules-polyfill` is doing automatically

**Comment by cyco130** [comment]
Do you have an example I can check that injects unnecessary polyfills? Maybe it'll help me understand what's exactly going on.

I built this plugin out of frustration with the other two you mentioned so I'm sure we can come up with something :)

**Comment by TrySound** [comment]
The the issue I attached above. Esbuild cannot require module polyfill from commonjs so it unconditionally injects everything in every commonjs.

> Install esbuild, add esbuild-plugin-polyfill-node plugin. Import commonjs module from entry point. You'll see all polyfills to be included in resulting bundle.
